src/index.js

```diff
@@ -17,18 +18,46 @@ export function createCycleMiddleware(mainFn, drivers) {
 
         return xs.create({
           start: listener => {
-            mainListener = listener;
+            actionListener = listener;
           },
           stop: () => {},
         });
       }
 
-      drivers.ACTION = cycleDriver;
+      const isSame = {};
+      const getCurrent = store.getState;
+      function stateDriver() {
+        return xs.create({
+          start: listener => {
+            stateListener = listener;
+          },
+          stop: () => {},
+        })
+        .fold((prevState, currState) => {
+          if (prevState === getCurrent) {
+            prevState = getCurrent();
+          }
+          if (prevState === currState) {
+            return isSame;
+          }
+          return currState;
+        }, getCurrent)
+        .map(state => state === getCurrent ? getCurrent() : state)
```

Comment:
This is needed to make sure that the `seed` ends up being the current value of the state when the stream is *created*, and not when the driver is initialized; since the stream is initialized when it's first used via `sources.STATE`, and not when the driver is initialized.